src/main/java/semantics/RDFImport.java

```diff
@@ -336,6 +336,67 @@ private int getHandleMultivalAsInt(String ignoreUrlsAsText) {
 
   }
 
+  @Procedure(mode = Mode.WRITE)
+  public Stream<DeleteResults> deleteRDF(@Name("url") String url, @Name("format") String format,
+      @Name(value = "params", defaultValue = "{}") Map<String, Object> props) {
+
+    final int handleVocabUris = (props.containsKey("handleVocabUris") ? getHandleVocabUrisAsInt(
+        (String) props.get("handleVocabUris")) : 0);
+    final boolean applyNeo4jNaming = (props.containsKey("applyNeo4jNaming") ? (boolean) props
+        .get("applyNeo4jNaming") : false);
+    final int handleMultival = (props.containsKey("handleMultival") ? getHandleMultivalAsInt(
+        (String) props.get("handleMultival")) : 0);
+    final List<String> multivalPropList = (props.containsKey("multivalPropList")
+        ? (List<String>) props.get("multivalPropList") : null);
+    final List<String> predicateExclusionList = (props.containsKey("predicateExclusionList")
+        ? (List<String>) props.get("predicateExclusionList") : null);
+    final List<String> customDataTypedPropList = (props.containsKey("customDataTypedPropList")
+        ? (List<String>) props.get("customDataTypedPropList") : null);
+    final boolean typesToLabels = (props.containsKey("typesToLabels") ? (boolean) props
+        .get("typesToLabels") : DEFAULT_TYPES_TO_LABELS);
+    final boolean keepLangTag = (props.containsKey("keepLangTag") ? (boolean) props
+        .get("keepLangTag") : false);
+    final boolean keepCustomDataTypes = (props.containsKey("keepCustomDataTypes") ? (boolean) props
+        .get("keepCustomDataTypes") : DEFAULT_KEEP_CUSTOM_DATA_TYPES);
+    final long commitSize = (props.containsKey("commitSize") ? (long) props.get("commitSize")
+        : DEFAULT_COMMIT_SIZE);
+    final long nodeCacheSize = (props.containsKey("nodeCacheSize") ? (long) props
+        .get("nodeCacheSize") : DEFAULT_NODE_CACHE_SIZE);
+    final String languageFilter = (props.containsKey("languageFilter") ? (String) props
+        .get("languageFilter") : null);
+
+    DeleteResults deleteResults = new DeleteResults();
+
+    DirectStatementDeleter statementDeleter = new DirectStatementDeleter(db,
+        (commitSize > 0 ? commitSize : 5000),
+        nodeCacheSize, handleVocabUris, handleMultival,
+        (multivalPropList == null ? null : new HashSet<String>(multivalPropList)),
+        keepCustomDataTypes,
+        (customDataTypedPropList == null ? null : new HashSet<String>(customDataTypedPropList)),
+        (predicateExclusionList == null ? null : new HashSet<String>(predicateExclusionList)),
+        typesToLabels, keepLangTag,
+        languageFilter, applyNeo4jNaming, log);
+    try {
+      checkIndexesExist();
+
+      InputStream inputStream = getInputStream(url, props);
+      RDFParser rdfParser = Rio.createParser(getFormat(format));
+      rdfParser.setRDFHandler(statementDeleter);
+      rdfParser.parse(inputStream, url);
+    } catch (MalformedURLException e) {
```

Comment:
Why is this exception not part of the next catch?
Shouldn't the status be KO as well in this case?